Fixes #3375 Added GDPR Agreement checkbox to contact form (incl. Lans)

SimSync opened the pull request:
This PR includes:
- Added update routine from 2.1.9 to 2.2.0 (because of a new pref).
- Added new pref for a privacy policy url in contact forms (incl. Lans) 
- Added GDPR Agreement checkbox to contact form (incl. Lans)

CaMer0n commented on e107_admin/update_routines.php:
```diff
@@ -564,6 +566,21 @@ function update_core_database($type = '')
+		return $just_check;
+	}
+
+
```
We don't actually need this, we just add the pref to default.xml and e107 will do the rest.

CaMer0n commented:
You're reading my mind with this change. :+1:  I believe we need a pref for 'terms and conditions' and another for 'privacy policy'. 

https://www.econsultancy.com/blog/69253-gdpr-10-examples-of-best-practice-ux-for-obtaining-marketing-consent

SimSync commented:
What about adding a new pref section "GDPR" with (currently) 2 prefs:
- url to privacy policy
- url to terms and conditions 

Than I would move the privacy policy from contact to the new section and we can put all GDPR related settings that may come in the new section.

SimSync commented:
Just implemented a new preferences section for GPDR related settings.
Moved "Privacy Policy" setting to the new section and added pref for "Terms and conditions"
Shortcode for comment form adapted to new pref name.

CaMer0n commented:
Looks great! :+1:  My only issue is that some people may have no idea what GDPR is. Perhaps "Legal" or something similar would be a better pref section title?

SimSync commented:
I would never look for "Legal" if i search for GDPR settings ....
Maybe we should include a short introduction about GDPR including a link to wikipedia about this term. This way all languages are "supported".

CaMer0n commented:
@SimSync I realized there's just one issue with the notice on the contact page. The data isn't stored, it is simply emailed to the admin. I wonder if it is still required?
